gfx/drivers_shader/slang_process.cpp

```diff
@@ -0,0 +1,406 @@
+      }
+      else
+      {
+         vs_compiler = new CompilerGLSL(output.vertex);
+         ps_compiler = new CompilerGLSL(output.fragment);
+      }
+
+      vs_resources = vs_compiler->get_shader_resources();
+      ps_resources = ps_compiler->get_shader_resources();
+
+      if (dst_type == RARCH_SHADER_HLSL || dst_type == RARCH_SHADER_CG)
+      {
+         CompilerHLSL::Options options;
+         CompilerHLSL*         vs = (CompilerHLSL*)vs_compiler;
+         CompilerHLSL*         ps = (CompilerHLSL*)ps_compiler;
+         options.shader_model     = version;
+         vs->set_options(options);
+         ps->set_options(options);
+
+         std::vector<HLSLVertexAttributeRemap> vs_attrib_remap;
+#if 0
+         /* remaps vertex shader output too so disable for now */
+         vs_attrib_remap.push_back({ 0, "POSITION" });
+         vs_attrib_remap.push_back({ 1, "TEXCOORD" });
+#endif
+         /* not exactly a vertex attribute but this remaps
+          * float2 FragCoord :TEXCOORD# to float4 FragCoord : SV_POSITION */
+         std::vector<HLSLVertexAttributeRemap> ps_attrib_remap;
+         for (Resource& resource : ps_resources.stage_inputs)
+         {
+            if (ps->get_name(resource.id) == "FragCoord")
+            {
+               uint32_t location = ps->get_decoration(resource.id, spv::DecorationLocation);
+               ps_attrib_remap.push_back({ location, "SV_Position" });
+            }
+         }
+         VariableTypeRemapCallback ps_var_remap_cb =
+               [](const SPIRType& type, const std::string& var_name, std::string& name_of_type) {
+                  if (var_name == "FragCoord")
+                     name_of_type = "float4";
+               };
+         ps->set_variable_type_remap_callback(ps_var_remap_cb);
+
+         vs_code = vs->compile(vs_attrib_remap);
+         ps_code = ps->compile(ps_attrib_remap);
+      }
+      else if (shader_info->type == RARCH_SHADER_GLSL)
+      {
+         CompilerGLSL::Options options;
+         CompilerGLSL*         vs = (CompilerGLSL*)vs_compiler;
+         CompilerGLSL*         ps = (CompilerGLSL*)ps_compiler;
+         options.version          = version;
+         ps->set_options(options);
+         vs->set_options(options);
+
+         vs_code = vs->compile();
+         ps_code = ps->compile();
+      }
+      else
+         goto error;
+
+      pass.source.string.vertex = (char*)malloc(vs_code.size() + 1);
+      memcpy(pass.source.string.vertex, vs_code.c_str(), vs_code.size() + 1);
+
+      pass.source.string.fragment = (char*)malloc(ps_code.size() + 1);
+      memcpy(pass.source.string.fragment, ps_code.c_str(), ps_code.size() + 1);
+
+      if (!slang_process_reflection(
+                vs_compiler, ps_compiler, vs_resources, ps_resources, shader_info, pass_number,
+                semantics_map, out))
+         goto error;
+
+   } catch (const std::exception& e)
+   {
+      RARCH_ERR("[slang]: spir2cross threw exception: %s.\n", e.what());
```

Comment:
SPIRV-Cross

Reply:
woops, copied that part from another file, typo included xD